trinity/main.py

```diff
@@ -79,21 +125,26 @@ def main():
         process.terminate()
 
 
-def ropsten_light_node_sync(log_queue, db_path):
-    """
-    Runs an LES node against the Ropsten network.
-    """
-    setup_queue_logging(log_queue)
+@with_queued_logging
+def run_chain(chain_config, sync_mode):
+    if not is_chain_initialized(chain_config):
+        # TODO: this will only work as is for chains with known genesis
+        # parameters.  Need to flesh out how genesis parameters for custom
+        # chains are defined and passed around.
+        chain_class = initialize_chain(chain_config, sync_mode=sync_mode)
+    else:
+        chain_class = get_chain_protocol_class(chain_config, sync_mode=sync_mode)
 
-    chaindb = BaseChainDB(LevelDB(db_path))
+    # TODO: this should probably be something that is passed in to allow
+    # specifying the db_path via the CLI as well as the db class.
```

Comment:
Seems like we should be choosing the class based on other configs they choose, like `--syncmode light`.

Reply:
I think this comment is outdated, but you are right that the class we choose will be based on various CLI options.